beacon_node/store/src/garbage_collection.rs

```diff
@@ -16,21 +17,22 @@ where
 
     /// Delete the temporary states that were leftover by failed block imports.
     pub fn delete_temp_states(&self) -> Result<(), Error> {
-        let delete_ops =
-            self.iter_temporary_state_roots()
-                .try_fold(vec![], |mut ops, state_root| {
-                    let state_root = state_root?;
-                    ops.push(StoreOp::DeleteState(state_root, None));
-                    Result::<_, Error>::Ok(ops)
-                })?;
-
-        if !delete_ops.is_empty() {
+        let mut ops = vec![];
+        self.iter_temporary_state_roots()?.for_each(|state_root| {
+            if let Ok(state_root) = state_root {
+                ops.push(state_root);
+            }
+        });
+        if !ops.is_empty() {
             debug!(
                 self.log,
                 "Garbage collecting {} temporary states",
-                delete_ops.len()
+                ops.len()
             );
-            self.do_atomically_with_block_and_blobs_cache(delete_ops)?;
+
+            self.delete_batch(DBColumn::BeaconState, ops.clone())?;
+            self.delete_batch(DBColumn::BeaconStateSummary, ops.clone())?;
+            self.delete_batch(DBColumn::BeaconStateTemporary, ops)?;
```

Comment:
Rather than cloning `ops` and doing 3 transactions, why don't we use `do_atomically`? Is it substantially slower?